.jenkins/pytorch/build.sh

```diff
@@ -150,6 +150,12 @@ if [[ "$BUILD_ENVIRONMENT" == *rocm* ]]; then
     export PATH="$CACHE_WRAPPER_DIR:$PATH"
   fi
 
+  # Set ROCM_ARCH to gtx900 unless defined otherwise
+  if [[ -z "$PYTORCH_ROCM_ARCH" ]]; then
+      echo "PYTORCH_ROCM_ARCH is undefined. Limiting it to gfx900"
+      export PYTORCH_ROCM_ARCH="gfx900"
+  fi
+
```

Comment:
```suggestion
  # Set ROCM_ARCH to gtx900 unless defined otherwise
  PYTORCH_ROCM_ARCH=${PYTORCH_ROCM_ARCH:-gfx900}
```

This could probably be written without the conditional

Reply:
Two issues.
1) our CI uses both gfx900 and gfx906 currently.  This would break some of our hosts.
2) This build.sh script is used by more than just CI jobs, i.e., for our own consumption as rocm pytorch developers, our QA teams, etc.  At the very least, please keep this conditional, otherwise we would need to hack at this script for all our internal uses.

Perhaps a different suggestion: can we detect that the script is being used by the circleci environment and conditionally set this to gfx900;gfx906 instead?  In short, keep the old default behavior unchanged?

Reply:
@jeffdaily this check already exists under `[[ $BUILD_ENVIRONMENT" == *rocm* ]]` which shouldn't be defined unless this code is invoked from CI

Reply:
Not so.  `BUILD_ENVIRONMENT` is set by the dockerfile build.  Many of our internal processes use the docker images built from either the .circleci/docker scripts or the jenkins docker scripts.  Both set BUILD_ENVIRONMENT as part of the builds.  That's why I suggested using a different env var that is only set by circleci.

Reply:
More clearly, `BUILD_ENVIRONMENT` becomes part of the running container and is not a CI-specific variable.

Reply:
@jeffdaily updated PR to use `CIRCLECI` environment variable.